react-components/src/components/Architecture/pointsOfInterest/PointsOfInterestSidePanel.tsx

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { CollapsablePanel } from '@cognite/cogs.js';
+import { type PropsWithChildren, type ReactElement, type ReactNode, useState } from 'react';
+import { useRenderTarget } from '../../RevealCanvas';
+import { PointsOfInterestTool } from '../../../architecture/concrete/pointsOfInterest/PointsOfInterestTool';
+import { type PointOfInterest } from '../../../architecture/concrete/pointsOfInterest/types';
+import styled from 'styled-components';
+import { useOnUpdate, useOnUpdateDomainObject } from '../useOnUpdate';
+import { type PointsOfInterestDomainObject } from '../../../architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject';
+import { PoIInfoPanelContent } from './PoIInfoPanelContent';
+import { PoIList } from './PoIList';
+
+export const PointsOfInterestSidePanel = ({ children }: PropsWithChildren): ReactNode => {
+  const renderTarget = useRenderTarget();
+
+  const tool = renderTarget.commandsController.getToolByType(PointsOfInterestTool);
+
+  const [enabled, setEnabled] = useState<boolean>(tool?.isEnabled ?? false);
+  const [poiObject, setPoiObject] = useState<PointsOfInterestDomainObject<any> | undefined>(
+    tool?.getPointsOfInterestDomainObject()
+  );
+
+  useOnUpdate(tool, () => {
+    if (tool === undefined) {
+      return;
+    }
+
+    setPoiObject(tool.getPointsOfInterestDomainObject());
+    setEnabled(tool.isChecked && tool.isEnabled);
+  });
+
+  return (
+    <CollapsablePanel
+      sidePanelRightVisible={enabled}
+      sidePanelRightWidth={500}
+      sidePanelRight={<PanelContainer poiObject={poiObject} />}>
+      {children}
+    </CollapsablePanel>
+  );
+};
+
+const PanelContainer = ({
+  poiObject
+}: {
+  poiObject?: PointsOfInterestDomainObject<any>;
+}): ReactNode => {
+  if (poiObject === undefined) {
+    return null;
+  }
+
+  return (
+    <PanelContentContainer>
+      <PanelContent poiObject={poiObject} />
+    </PanelContentContainer>
+  );
+};
+
+const PanelContent = ({
+  poiObject
+}: {
+  poiObject: PointsOfInterestDomainObject<any>;
+}): ReactElement => {
```

Comment:
Is it not better to use `ReactNode` here as well even not having any the other returning for now?